config/config.go

```diff
@@ -37,29 +37,37 @@ var (
 	}
 )
 
+var (
+	defaultArgsBlacklist = []string{"password", "passwd", "mysql_pwd", "access_token", "auth_token", "api_key", "apikey", "secret", "credentials", "stripetoken"}
+)
+
 type proxyFunc func(*http.Request) (*url.URL, error)
 
 // AgentConfig is the global config for the process-agent. This information
 // is sourced from config files and the environment variables.
 type AgentConfig struct {
-	Enabled       bool
-	APIKey        string
-	HostName      string
-	APIEndpoint   *url.URL
-	LogFile       string
-	LogLevel      string
-	QueueSize     int
-	Blacklist     []*regexp.Regexp
-	MaxProcFDs    int
-	ProcLimit     int
-	AllowRealTime bool
-	Transport     *http.Transport `json:"-"`
-	Logger        *LoggerConfig
-	DDAgentPy     string
-	DDAgentBin    string
-	DDAgentPyEnv  []string
-	StatsdHost    string
-	StatsdPort    int
+	Enabled              bool
+	APIKey               string
+	HostName             string
+	APIEndpoint          *url.URL
+	LogFile              string
+	LogLevel             string
+	QueueSize            int
+	Blacklist            []*regexp.Regexp
+	ArgsBlacklist        []*regexp.Regexp
+	DefaultArgsBlacklist []*regexp.Regexp
+	UseDefArgsBlacklist  bool
+	CustomArgsBlacklist  []*regexp.Regexp
```

Comment:
I would be in favor of encapsulating all the blacklisting behavior into a `DataScrubber` or `Blacklister` (and a separate module since this one is already rather large). Then your would have a simple interface like like `cfg.scrubber.ScrubCmdline(...)`.

As an aside I think using Sentry's "data scrubber" language is nicer than "blacklist" for something that will be customer visible